### PR TITLE
[WIP] Create Docker container for tests

### PR DESCRIPTION
**Pull request by fievelk**
# WIP

Docker container for tests

### CODE REVIEW

**Comment by fievelk** [comment]
@alvations I'm trying to set up a docker container to run the tests, so I want to build it using the correct dependencies.

I can see that the Travis pipeline is checking for a specific java version, in a specific location. Could you please give me more details about this? For example, why is this check needed? And why do we use this specific location? I know it's a pretty old commit, but I hope you can help :)
https://github.com/nltk/nltk/blob/d13078e0441cd46f46b41a9db70dbde0ffacd4fe/tools/travis/pre-install.sh#L8

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# USER ubuntu
+
+WORKDIR /usr/src/app
+
+# Copy source files from the host to the container's workdir
+COPY . .
+
+# Install NLTK package
+RUN pip install .
```
Comment (bsolomon1124) [inline Dockerfile:11]: To cut down on image layers, concatenate these into one:

```
RUN set -ex \
    && python -m pip install . tox \
    && . tools/travis/pre-install.sh \
    && chmod +x tools/travis/coverage-pylint.sh \
    && chmod +x tools/travis/third-party.sh
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN pip install tox
+
+RUN source tools/travis/pre-install.sh
```
Comment (bsolomon1124) [inline Dockerfile:14]: Use `.` over `source` for POSIX `sh` compliance

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+RUN chmod +x tools/travis/coverage-pylint.sh
+RUN chmod +x tools/travis/third-party.sh
+
+# Keep the container running
+CMD tail -f /dev/null
```
Comment (bsolomon1124) [inline Dockerfile:19]: If there is a need for this CMD in the first place, [JSON-style](https://github.com/hadolint/hadolint/wiki/DL3025) is usually preferable:

```
CMD ["tail", "-f", "/dev/null"]
```

File: .travis.yml [hunk 4]
```diff
@@ -1,52 +1,18 @@
...
 before_install:
-  - source tools/travis/pre-install.sh # Checks Java and Python versions.
-  - chmod +x tools/travis/coverage-pylint.sh
-  - chmod +x tools/travis/third-party.sh
-
-install:
-  - source tools/travis/install.sh
-  #- source tools/travis/third-party.sh
-  - pip install tox
+  - docker build --tag nltk .
```
Comment (bsolomon1124) [inline .travis.yml:13]: Putting the Dockerfile in the project root directory sends a large build context to the docker daemon.

Putting it in `docker/` or something like that will avoid that.

File: .travis.yml [hunk 5]
```diff
@@ -1,52 +1,18 @@
...
+  - docker run --detach --name nltk nltk
```
Comment (bsolomon1124) [inline .travis.yml:14]: You can probably avoid the `run --detach` altogether and just use this in the `script`:

```
docker container run -t --rm -e ${TOXENV} nltk tox -vv
```

(Not sure if I have that quite right, might need `sh -c "xxx"`.)

And change CMD to:

```
CMD ["/bin/bash"]
```

**Comment by fievelk** [comment]
Thank you @bsolomon1124, very appreciated suggestions! 
I am currently slowed down by a tendonitis, but I hope I'll be back working on this soon.

**Comment by stevenbird** [comment]
Hi @fievelk ... I hope you're better... looking forward to your return

**Comment by stevenbird** [comment]
Inactive, so closing. Happy to reconsider in future...

**Pull request closed without merging**